From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Tue, 7 Feb 2012 23:23:36 +0000
Subject: netxen_nic: signedness bug in netxen_md_entry_err_chk()
Patch-mainline: v3.4-rc1
Git-commit: a584b7ae4ef01425f6827657abdc3e4db49b3578
References: bnc#784815 FATE#313898

"esize" should be signed because it can be negative here.  For example,
when we call it in netxen_parse_md_template(), it could be -1 from the
return value of netxen_md_L2Cache().

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Acked-by: David Chang <dchang@suse.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/qlogic/netxen/netxen_nic_hw.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/qlogic/netxen/netxen_nic_hw.c
+++ b/drivers/net/ethernet/qlogic/netxen/netxen_nic_hw.c
@@ -2354,7 +2354,7 @@ netxen_md_rdqueue(struct netxen_adapter
 */
 
 static int netxen_md_entry_err_chk(struct netxen_adapter *adapter,
-				struct netxen_minidump_entry *entry, u32 esize)
+				struct netxen_minidump_entry *entry, int esize)
 {
 	if (esize < 0) {
 		entry->hdr.driver_flags |= NX_DUMP_SKIP;
